src/EFCore.PG/Query/Internal/NpgsqlSqlTranslatingExpressionVisitor.cs

```diff
@@ -246,6 +253,52 @@ protected override Expression VisitMethodCall(MethodCallExpression methodCall)
                 }
             }
 
+            {
+                // Pattern match for: array.Any(e => e == x) (and other equality patterns)
+                // Transform this to Contains.
+                if (method.IsClosedFormOf(EnumerableAnyWithPredicate) &&
+                    arguments[1] is LambdaExpression wherePredicate &&
+                    TryMatchEquality(wherePredicate.Body, out var left, out var right) &&
+                    (left == wherePredicate.Parameters[0] || right == wherePredicate.Parameters[0]))
+                {
+                    var item = left == wherePredicate.Parameters[0]
+                        ? right
+                        : right == wherePredicate.Parameters[0]
+                            ? left
+                            : null;
+
+                    return item is null
+                        ? null
+                        : Visit(Expression.Call(Contains.MakeGenericMethod(method.GetGenericArguments()[0]), array, item));
+                }
+
+                static bool TryMatchEquality(Expression expression, out Expression left, out Expression right)
+                {
+                    if (expression is BinaryExpression binary)
+                    {
+                        (left, right) = (binary.Left, binary.Right);
+                        return true;
+                    }
+                    if (expression is MethodCallExpression methodCall)
+                    {
+                        if (methodCall.Method == ObjectEquals)
+                        {
+                            (left, right) = (methodCall.Arguments[0], methodCall.Arguments[1]);
+                            return true;
+                        }
+
+                        if (methodCall.Method.Name == nameof(object.Equals) && methodCall.Arguments.Count == 1)
+                        {
+                            (left, right) = (methodCall.Object, methodCall.Arguments[0]);
+                            return true;
+                        }
+                    }
+
+                    (left, right) = (null, null);
+                    return false;
```

Comment:
If there would be a way to express such a thing in one line, it would be fine, but not now. In C++ which you dislike it's possible thankfully to comma operator, so here is an idea for a proposal for C# language team:

```csharp
return (left = null, right = null, false); // only the last value used as the result
```

In that case you would be able to rewrite the whole method as an expression. Do you like it?

Reply:
The problem is that your expression is already legal in C#, and means something else:

```c#
public (string, string, string) Bar(ref string left, ref string right)
{
    return (left = null, right = null, "hello");
}
```

Reply:
If the left side expects less values than the right, it should work:

```csharp
public string Brar(ref string left, ref string right) =>
    (left = null, right = null, "hello");
```

Otherwise, it would be cool to specify `return` keyword to solve the problem:

```csharp
public string Brar(ref string left, ref string right) =>
    (left = null, right = null, return "hello");
```

Reply:
IMHO that is super ambiguous with the tuple syntax... There's already a `(a, b, c)` which means something very clear. In your first example, you cannot know the type of that expression without looking at the method return type, which I think is a no-no in language design (even if we have target-typed assignments now)...

Reply:
In case of expression based methods it could be fine since the type is nearby, and last example includes `return`.